gcc/rust/rust-session-manager.h

```diff
@@ -267,6 +271,12 @@ struct Session
   void parse_files (int num_files, const char **files);
   void init_options ();
 
+  inline const char *include_extra_file (std::string filename)
+  {
+    extra_files.push_back (std::move (filename));
+    return extra_files.back ().c_str ();
+  }
```

Comment:
I think this is a great idea. The API might be a little weird but I am unsure about what would be a better solution. We should open up an issue to also use this for module loading.

For now, I think it would be good if this function had a little more documentation. Something explaining that it returns a raw pointer to data stored in the session manager instance for example

Reply:
> The API might be a little weird but I am unsure about what would be a better solution.

This is due to an issue of unknown cause I discovered when I was implementing this workaround. If you return a `std::string&`, then when you pass its `.c_str()` pointer later to `Lexer`, the pointer will point to an invalid stack address for unknown reasons. If you return a `const char*` pointer directly, it will return a heap address pointing directly to the target string.

I tried `asan` and `valgrind`, and surprisingly, they were unable to find anything.

Reply:
I don't know what your code was like, but if the reference you were returning referred to the moved value then it would make sense since the string would now be in an erroneous state? Or were you doing something like `return extra_files.back ()` too?

Regardless, I don't think it's especially bad that this returns a raw C pointer. But I think it should be documented so that we know the actual data is stored inside the session manager and someone using `include_extra_file` does not have to free the pointer they get or something like that.

Reply:
> I don't know what your code was like, but if the reference you were returning referred to the moved value then it would make sense since the string would now be in an erroneous state? Or were you doing something like `return extra_files.back ()` too?

Yes, I was using `return extra_files.back ()` because obviously `move`d data can't be used anymore.

> 
> Regardless, I don't think it's especially bad that this returns a raw C pointer. But I think it should be documented so that we know the actual data is stored inside the session manager and someone using `include_extra_file` does not have to free the pointer they get or something like that.

Comments added.